codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -492,13 +476,12 @@ public void onGoAwayRead(ChannelHandlerContext ctx, int lastStreamId, long error
         public void onWindowUpdateRead(ChannelHandlerContext ctx, int streamId, int windowSizeIncrement)
                 throws Http2Exception {
             Http2Stream stream = connection.requireStream(streamId);
-            verifyGoAwayNotReceived();
-            if (stream.state() == CLOSED || shouldIgnoreFrame(stream, false)) {
-                // Ignore frames for any stream created after we sent a go-away.
+
+            if (stream.state() == CLOSED || streamCreatedAfterGoAwaySent(stream)) {
```

Comment:
@Scottmitch I really don't think we'll ever care about the flow control windows for closed streams ... even with extension frames.  We only care about the connection window getting updated properly (which would never fall into this `if` statement).